pkg/state/storage.go

```diff
@@ -30,7 +31,17 @@ func NewStorage(forFile string, logger *zap.SugaredLogger, glob func(string) ([]
 func (st *Storage) resolveFile(missingFileHandler *string, tpe, path string) ([]string, bool, error) {
```

Comment:
> This enhancement allows for referencing remote (go-getter) files to be
fetched, cached and referenced.

Loading env values from remote already supported, [link][1]. [Here][2] is a code that is responsible for that. 

Does it mean that you propose to trait any file link as a potentially remote file? In that I think `LoadEnvironmentValues` function should be adjusted as well.

[1]: https://github.com/helmfile/helmfile#loading-remote-environment-values-files
[2]: https://github.com/helmfile/helmfile/blob/45be24da53dd8af3342b9946044b687b26ce9a8a/pkg/state/envvals_loader.go#L44-L55

Reply:
Thank you for pointing out. The PR solves the same issues ( roboll/helmfile#1838 / roboll/helmfile#1499 ) as your RP #42. The title of the PR was misleading and I changed it from environment to release. I also added a basic example to illustrate the enhancement.

The difference of the two implementations is, that this PR also fetches secret values files. Since your PR only patches `generateVanillaValuesFiles`.
Maybe we find a middle ground and we can combine the two options.

Reply:
Yes, thanks for the additional input!

Yeah, what I'm pointing is that in your case resolving happens twice. Maybe it is better to patch vanilla values and secrets fletcher separately. Not the base function.

Or, as a common solution better than repeating solution 3 times it makes sense to remove env vars loader and just use your patch everywhere.

Reply:
> in your case resolving happens twice

If we're going to support this remote file loading universally within helmlfile, it might make sense to patch resolveFile this way. Let's merge this and see how it goes.

I'm unsure if I'm aware of all the consequences of doing it though. I'd appreciate it if you could submit follow-up PRs to duplicate the remote file fetching step :bow: